app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulation.java

```diff
@@ -0,0 +1,169 @@
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Sets;
+
+/**
+ * A sub-population of the study participants who will receive a unique consent based on selection by a data group, 
+ * an application version, or both. 
+ */
+@DynamoDBTable(tableName = "Subpopulation")
+@BridgeTypeName("Subpopulation")
+public final class DynamoSubpopulation implements Subpopulation {
+
+    private String studyIdentifier;
+    private String guid;
+    private String name;
+    private String description;
+    private boolean required;
+    private boolean deleted;
+    private Integer minAppVersion;
+    private Integer maxAppVersion;
+    private Long version;
+    private Set<String> allOfGroups;
+    private Set<String> noneOfGroups;
+
+    public DynamoSubpopulation() {
+        this.allOfGroups = Sets.newHashSet();
+        this.noneOfGroups = Sets.newHashSet();
+    }
+    
+    @JsonIgnore
+    @Override
+    @DynamoDBHashKey
+    public String getStudyIdentifier() {
+        return studyIdentifier;
+    }
+    @Override
+    public void setStudyIdentifier(String studyIdentifier) {
+        this.studyIdentifier = studyIdentifier;
+    }
+    @Override
+    @DynamoDBRangeKey
+    public String getGuid() {
+        return guid;
+    }
+    @Override
+    public void setGuid(String guid) {
+        this.guid = guid;
+    }
+    @DynamoDBAttribute
+    @Override
+    public String getName() {
+        return name;
+    }
+    @Override
+    public void setName(String name) {
+        this.name = name;
+    }
+    @DynamoDBAttribute
+    @Override
+    public String getDescription() {
+        return description;
+    }
+    @Override
+    public void setDescription(String description) {
+        this.description = description;
+    }
+    @DynamoDBAttribute
+    @Override
+    public boolean isRequired() {
+        return required;
+    }
+    @Override
+    public void setRequired(boolean required) {
+        this.required = required;
+    }
+    @DynamoDBAttribute
+    @Override
+    public boolean isDeleted() {
+        return deleted;
+    }
+    @Override
+    public void setDeleted(boolean deleted) {
+        this.deleted = deleted;
+    }
+    @DynamoDBMarshalling(marshallerClass = StringSetMarshaller.class)
+    @DynamoDBAttribute
+    @Override
+    public Set<String> getAllOfGroups() {
+        return allOfGroups;
+    }
+    public void setAllOfGroups(Set<String> dataGroups) {
+        this.allOfGroups = (dataGroups == null) ? Sets.newHashSet() : dataGroups;
+    }
```

Comment:
Should the StringSet setters also be part of the interface? All the other setters are. Otherwise, it seems odd that the only way to set StringSets are parsing them from JSON.

Also, should we make a copy of the data group? Otherwise, weird things might happen if someone changes the original string set out from under us.